src/core.js

```diff
@@ -4,7 +4,7 @@ import cookies from "./cookies.js";
 let core = {
 
     versions: {
-        libVersion: "1.12.5454545454-build-number", //"5454545454-build-number" will be replaced by teamcity. also in package.json
+        libVersion: "___REPLACED DURING BUILD JOB___",
```

Comment:
This could well be a silly question, revealing my ignorance of JS, but is it possible for someone to run the connector without "building" it (like during development), and thus actually send this placeholder as the `Lib-Version` header? Maybe it's safer go the other way and have the version copied from here to package.json?

Reply:
In a Node.js setting this is possible.
Previously, this kind of usage would have sent `1.12.5454545454-build-number`. Is this less of an issue?

Generally speaking, I think `package.json` should be the source of the version number.
It is possible to dynamically read the version from `package.json` at runtime.
```js
{ libVersion: require("package.json").version }
```
For a Node bundle this might be fine.
But for a browser bundle this would have to get inlined. I tried this, but our current bundler simply puts the entire `package.json` into the final bundle and reads from it like a module. We can't do that because it wastes space and reveals the libraries dependencies and their versions.

Various solutions thinkable here. Which one do you prefer?

Reply:
I think receiving `1.12.5454545454-build-number` as `Lib-Version` is not as bad, because then we still basically know the source code version.

OK, so if you want to have the version in package.json, and it's too hard to read it from package.json, then another option is to have it in both places (as before), but have the build fail if the versions differ. Maybe it's not so bad to have to update the version number in two places, as long as we ensure they stays in sync?

Reply:
Sorry for the late reply. I don't think we need too much automation, given how rarely we actually update the JS connector. Personally I'd be happy with manually updating the version in two places, as long as we somehow prevent the two from getting out of sync. But if you're going to be the one maintaining this then it's really up to you!